File: コンテスト/AtCoder Regular Contest 147/A - Max Mod Min/max_mod_min.py
#!/usr/bin/env python3

from collections import deque


if __name__ == "__main__":
    n = int(input())
    a = list(map(int, input().split()))

    a.sort()
    a = deque(a)
    ans = 0
    while len(a) > 1:
        curr = a.pop()
        if curr % a[0] > 0:
            a.appendleft(curr % a[0])
        ans += 1

    print(ans)

File: コンテスト/AtCoder Regular Contest 147/A - Max Mod Min/max_mod_min.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	in := bufio.NewReader(os.Stdin)

	var n int
	fmt.Fscan(in, &n)

	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(in, &a[i])
	}

	sort.Ints(a)
	ans := 0
	for a[0] > 1 {
		curr := a[len(a)-1] % a[0]
		if curr == 0 {
			a = a[:len(a)-1]
		} else {
			a = a[:len(a)-1]
			a = append([]int{curr}, a...)
		}
		ans++
	}

	fmt.Println(ans + len(a) - 1)
}
